### ISSUE TITLE
Prevent Solver to re-use a word already on Grid

### ISSUE THREAD

**Issue by guillaumepotier** [jsgonsette/Wizium#12]
Hi there,

First of all, thanks a lot for this great project!

I'd like to add a feature/option in the Solver, to avoid picking again a word already used in the Grid. I first thought to create a Dictionary.removeEntry once a word is put in the Grid but it won't do because of the generating process, the word in the Grid could be removed later if we find a dead-end, and without it in the Dictionary anymore it could be a mess..

Do you have any idea on how to implement that?

Maybe a forbiddenMask/forbiddenWords on the findEntry method?

Thanks a lot for your help.

Best

**Comment by jsgonsette** [comment]
Hello, 
this issue has already been discussed here:
https://github.com/jsgonsette/Wizium/issues/6

where I proposed some steps to correct this problem.
I think this could be a very nice improvement if you are comfortable implementing it :-)
Kind regards
JS

**Comment by guillaumepotier** [comment]
Hi @jsgonsette 

Thanks for your answer, I found #6 yesterday evening after having posted this issue, sorry for that.

I'll try my best since I've not written C++ since high school a while back. I've had a hard time yesterday already to figure out how things were done. I'll obviously submit a PR if I manage to add this improvement if I do not succumb meanwhile to the tentation to write my own node javascript library for my very basic needs, ie. only generate 5x5 static grids with defined block cells.

FYI, we used your lib for generating grids of [Motif](https://playmotif.fr/)
We mentioned your and your project in the [About](https://playmotif.fr/#about) section

Keep you posted

Best
